### PR TITLE
Added a local HTTP server that acts similar to S3 to serve files created by Theater.

### PR DESCRIPTION
**Pull request by jmkulwik**
Currently we have divergent logic between local theater and non-local theater. On non-local environments we save the images and audio to S3 and retrieve them via url on the client side. Locally we send back the raw bytes. 

This PR aligns that logic by setting up a minimal HTTP server that serves GET requests to the local file system. It also cleans up the code that was maintaining the divergent logic.

There will be a follow-up change on the client side to remove the listener for the binary image and audio files.
Follow-up change: https://github.com/code-dot-org/code-dot-org/pull/42114

Jira Task: [CSA-576](https://codedotorg.atlassian.net/browse/CSA-576)

### CODE REVIEW

File: org-code-javabuilder/lib/src/main/java/dev/javabuilder/HttpFileServer.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  @Override
+  protected void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+    // https://docs.oracle.com/javaee/5/api/javax/servlet/ServletResponse.html
+    // https://docs.oracle.com/javaee/5/api/javax/servlet/http/HttpServletResponse.html
+    // NOTE: This is _NOT_ a safe method of handling requests from a client. We are naively serving
+    // whatever file is asked for by the user. This should _ONLY_ be used for local development.
+    String[] urlParts = request.getRequestURI().split("/");
+    String fileName = urlParts[urlParts.length - 1];
+    File file = new File(fileName);
```
[the remark this answers is not in the document]
Comment (jmkulwik) [reply org-code-javabuilder/lib/src/main/java/dev/javabuilder/HttpFileServer.java:32]: It just ends up in the runtime directory. tmp would probably be better. I'll move it there instead.

[the remark this answers is not in the document]
Comment (jamescodeorg) [reply org-code-javabuilder/lib/src/main/java/dev/javabuilder/HttpFileServer.java:32]: If it's not too difficult to put it in a subdirectory of tmp specific to this scenario, that would probably be even safer.

File: org-code-javabuilder/lib/src/main/java/dev/javabuilder/HttpFileServer.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This sets up an HTTP server for local development when the client needs to be able to access
+ * files generated by Javabuilder (such as when using Theater). In production, this is handled by
+ * S3.
+ */
+@WebServlet(name = "FileServlet", urlPatterns = "/files/*")
+public class HttpFileServer extends HttpServlet {
```
[the remark this answers is not in the document]
Comment (jmkulwik) [reply org-code-javabuilder/lib/src/main/java/dev/javabuilder/HttpFileServer.java:27]: I think I may have mitigated the issue by limiting to only the two filenames. Does this look good now?

File: org-code-javabuilder/lib/src/main/java/dev/javabuilder/HttpFileServer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmkulwik) [reply org-code-javabuilder/lib/src/main/java/dev/javabuilder/HttpFileServer.java]: Good call. Done :)

**Review by jamescodeorg — APPROVED** [review]

**Review by sanchitmalhotra126 — APPROVED** [review]
Awesome, LGTM!

**Pull request merged**